src/Confluent.Kafka/Producer.cs

```diff
@@ -790,6 +804,22 @@ internal Producer(ProducerBuilder<TKey, TValue> builder)
         ///     A delegate that will be called with a delivery report corresponding
         ///     to the produce request (if enabled).
         /// </param>
+        /// <exception cref="ProduceException{TKey,TValue}">
+        ///     Thrown in response to any produce request error that is known
```

Comment:
"produce request" is somewhat problematic here since these errors are raised prior to any produce request.

Probably also want to say that these are local errors, not broker.

Reply:
yep good point. i didn't include the point about local errors since i think this is implied by 'known immediately' and i'm in a hurry :-)